### DISCUSSION TITLE
How to create Decimal Arrays?

### DISCUSSION THREAD

**Discussion by finlaydotb** [apache/arrow-rs#5816] [Q&A] [answered]
So for example the [Decimal128](https://docs.rs/dec/latest/dec/struct.Decimal128.html) type. 

The example 

```
// Create from Vec<Option<i18>>
let arr = Decimal128Array::from(vec![Some(1), None, Some(2)]);
// Create from Vec<i128>
let arr = Decimal128Array::from(vec![1, 2, 3]);
// Create iter/collect
let arr: Decimal128Array = std::iter::repeat(42).take(10).collect();
```

But I do not want to use `Decimal128Array` to create an array of integers. 

So my main requirement is that I want to create arrays of decimal values. I do not want to use float because some of the operations exposed via floats are not deterministic.

So basically how do I create decimal arrays?

**Comment by tustvold** [comment, marked as answer]
Decimals are just integers multiplied by some scale. So if your scale is 2, `1.01` would be the integer `101`

**Reply by finlaydotb** [reply]
Thanks. It was not just clear from the API, how to specify the scale while creating, or even how to access the scale from a created value...which I think would have been useful in the example given that when decimal needs to be created, it is 99% going to be defined with a scale.
